go-controller/pkg/networkmanager/network_controller.go

```diff
@@ -245,13 +245,32 @@ func (c *networkController) syncAll() error {
 	start := time.Now()
 	klog.Infof("%s: syncing all networks", c.name)
 	for _, network := range validNetworks {
-		err := c.syncNetwork(network.GetNetworkName())
+		// Ensure node is allocated with host subnet for the network, When subnet is not allocated
+		// for the node then skip syncing the network and proceed with syncing remaining network.
+		// Return the error at the end when k8s.ovn.org/node-subnets annotation is not set for a
+		// network and let the caller to decide on what to do with it.
+		networkName := network.GetNetworkName()
+		if c.node != "" && util.IsNetworkSegmentationSupportEnabled() && network.IsPrimaryNetwork() &&
+			network.TopologyType() == types.Layer3Topology {
+			node, err := c.nodeLister.Get(c.node)
+			if err != nil {
+				return fmt.Errorf("error retrieving node %s object while syncing network %s: %w",
+					c.node, networkName, err)
+			}
+			_, err = util.ParseNodeHostSubnetAnnotation(node, networkName)
+			if err != nil && util.IsAnnotationNotSetError(err) {
+				klog.Errorf("Ignoring syncing network %s because k8s.ovn.org/node-subnets annotation is not set on node %s",
+					networkName, c.node)
+				continue
+			}
+		}
```

Comment:
Do you think we can move this block to a separate function, like `IsNetworkReadyForSync` or a similar name?

Reply:
sure, moved this new block into a `isNetworkReadyForSync`  method.